### PR TITLE
100 forms

### PR DESCRIPTION
**Pull request by agentrickard**
Working integration of user-specific forms and permissions, with tests.

### CODE REVIEW

File: src/Form/AssignUserForm.php [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    // Load all schemes.
+    $schemes = $this->schemeStorage->loadMultiple();
+    foreach ($schemes as $scheme) {
+      $user_sections = $this->userSectionStorage->getUserSections($scheme, $user, FALSE);
+      $admin_sections = $this->userSectionStorage->getUserSections($scheme, $account, FALSE);
+      $options = $this->getFormOptions($scheme);
+      $role_sections = $this->roleSectionStorage->getRoleSections($scheme, $user);
+      foreach ($options as $value => $label) {
+        if (in_array($value, $role_sections)) {
```
Comment (larowlan) [inline src/Form/AssignUserForm.php:127]: this should use the third argument, as 0 would match any string

File: src/Form/AssignUserForm.php [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  }
+
+  /**
+   * Checks access for the form.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function checkAccess() {
+    $permissions = ['assign workbench access', 'assign selected workbench access'];
```
Comment (larowlan) [inline src/Form/AssignUserForm.php:208]: I think routing.yml supports a `+` format for multiple permissions

File: tests/src/Functional/AssignUserFormTest.php [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    $existing_users = $user_storage->getEditors($taxonomy_scheme, 2);
+    $this->assertEquals($expected, array_keys($existing_users));
+    $existing_users = $user_storage->getEditors($taxonomy_scheme, 3);
+    $this->assertEquals([], array_keys($existing_users));
+
+    // Check that the $none_user cannot access the page.
+    $this->drupalLogin($none_user);
+    $this->drupalGet(Url::fromRoute('entity.section_association.edit', ['user' => $none_user->id()]));
+    $assert = $this->assertSession();
+    $assert->statusCodeEquals(403);
```
Comment (larowlan) [inline tests/src/Functional/AssignUserFormTest.php:253]: nice

File: src/Form/AssignUserForm.php (hunk 2 above)
Comment (agentrickard) [reply src/Form/AssignUserForm.php:208]: It does! fixed.

File: src/Form/AssignUserForm.php (hunk 1 above)
Comment (agentrickard) [reply src/Form/AssignUserForm.php:127]: Done and testing correctly.

**Pull request merged**